src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -745,7 +745,7 @@ private AbstractType<?> collectionCellValueType(AbstractType<?> type, IndexTarge
                         return CompositeType.getInstance(collection.nameComparator(), collection.valueComparator());
                 }
             default:
-                throw new IllegalArgumentException("Unsupported collection type: " + collection.kind);
+                throw new IllegalArgumentException("Unsupported collection type: " + collection.kind + "; index type: " + indexType.name());
```

Comment:
unrelated to the patch, more a product of the fuzz testing...

`RandomSchemaV2Test` now does far more randomness than before, includes accord, *and* tests indexes...

This test is failing due to accord blocking indexes (which is a bug)... Celeb and Ariel are looking into this